src/FiberMap.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace React\Async;
+
+use Fiber;
+use React\Promise\PromiseInterface;
+
+/**
+ * @internal
+ */
+final class FiberMap
+{
+    private array $status = [];
+    private array $map   = [];
+
+    public function register(Fiber $fiber): void
+    {
+        $this->status[spl_object_hash($fiber)] = false;
+        $this->map[spl_object_hash($fiber)] = [];
+    }
+
+    public function cancel(Fiber $fiber): void
+    {
+        $this->status[spl_object_hash($fiber)] = true;
+    }
+
+    public function isCanceled(Fiber $fiber): bool
```

Comment:
nit: should be `isCancelled()` (british spelling for consistency throughout our components)